setup.py

```diff
@@ -508,13 +604,105 @@ def setup_pytorch_extension() -> setuptools.Extension:
         name="transformer_engine_extensions",
         sources=sources,
         include_dirs=include_dirs,
-        # libraries=["transformer_engine"], ### TODO (tmoon) Debug linker errors
         extra_compile_args={
             "cxx": cxx_flags,
             "nvcc": nvcc_flags,
         },
+        **lib_kwargs
     )
 
+def setup_jax_extension() -> setuptools.Extension:
+    """Setup PyBind11 extension for JAX support"""
+    # Source files
+    src_dir = root_path / "transformer_engine" / "jax" / "csrc"
+    sources = [
+        src_dir / "extensions.cpp",
+        src_dir / "modules.cpp",
+        src_dir / "utils.cu",
+    ]
+
+    # Header files
+    include_dirs = [
+        CUDA_HOME / "include",
+        CUDNN_INCLUDE,
+        root_path / "transformer_engine" / "common" / "include",
+        root_path / "transformer_engine" / "jax" / "csrc",
+        root_path / "transformer_engine",
+    ]
+
+    # Compile flags
+    cxx_flags = [ "-O3" ]
+    nvcc_flags = [
+        "-O3",
+        "--forward-unknown-opts",
+    ]
+    if GLIBCXX_USECXX11_ABI:
+        # Core TE library was built with C++11 ABI (for PyTorch compat) so we need to do the same
+        # for all the framework extensions.
+        cxx_flags.append("-D_GLIBCXX_USE_CXX11_ABI=1")
+        nvcc_flags.append("-D_GLIBCXX_USE_CXX11_ABI=1")
+
+    # Linked libraries
+    libraries = [
+        'cudart',
+        'cublas',
+        'cublasLt',
+        'cudnn',
+    ]
+    lib_kwargs = {}
+    if lib_extension() == 'dll':
+        # Windows needs explicit file paths for each library.
+        cuda_lib_dir = CUDA_HOME / 'lib' / 'x64'
+        libraries = [ cuda_lib_dir / f'lib{name}.dll' for name in libraries ]
+
+        # Core TE library is in the same folder as the extension library so it doesn't
+        # need an absolute path, just the full file name.
+        lib_kwargs['libraries'] = [ str(path) for path in libraries ] + \
+                                  [ 'libtransformer_engine.dll' ]
+    else:
+        # Set link and runtime paths for CUDA libraries.
+        cuda_lib_dir = CUDA_HOME / 'lib64'
+        if (not os.path.exists(cuda_lib_dir) and os.path.exists(CUDA_HOME / 'lib')):
```

Comment:
Nit:
```suggestion
        if (not cuda_lib_dir.exists() and (CUDA_HOME / 'lib').exists()):
```
`os.path` and Pathlib basically have the same functionality, but it's nicer to be consistent with which one you use.